internal/http_api/api_request.go

```diff
@@ -19,12 +19,10 @@ type deadlinedConn struct {
 }
 
 func (c *deadlinedConn) Read(b []byte) (n int, err error) {
-	c.Conn.SetReadDeadline(time.Now().Add(c.Timeout))
 	return c.Conn.Read(b)
 }
 
 func (c *deadlinedConn) Write(b []byte) (n int, err error) {
-	c.Conn.SetWriteDeadline(time.Now().Add(c.Timeout))
```

Comment:
Now that we've removed these, we need to set `ResponseHeaderTimeout` to our configured request timeout on the `http.Transport` instantiation below on line 30. `NewDeadlineTransport` will need to take both connect timeout and request timeout.